src/movies.js

```diff
@@ -1,16 +1,89 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies){
+    let answer1 = movies.map((movie) => {
+        return movie.director;
+    })
+return answer1;
+};
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    let drama;
+    let spiel = movies.filter((movie) => {
+    for (let i=0; i < movie.genre.length; i++){
+       drama = movie.genre[i] == "Drama" && movie.director == "Steven Spielberg";
+       if(drama){break;}
+    } 
+    return drama
+}) 
+return spiel.length;
+};
+
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+
+
+function ratesAverage(arrMovies) {
+    let sum = arrMovies.reduce(function(acc, elem){
```

Comment:
to check the jasmine test to return 0 if there is no movies you could start with a simple if condition that checks that and breaks the function by returning 0. (When a function encounters a return if will stop running the rest of the code inside.